Funcky.SourceGenerator/OrNoneFromTryPatternGenerator.cs

```diff
@@ -40,69 +37,134 @@ private static IncrementalValueProvider<ImmutableArray<MethodPartial>> GetOrNone
         => context.SyntaxProvider.CreateSyntaxProvider(predicate: IsSyntaxTargetForGeneration, transform: GetSemanticTargetForGeneration)
             .WhereNotNull()
             .Combine(context.CompilationProvider)
-            .Select((state, cancellationToken) => ToMethodPartial(state.Left, state.Right, context, cancellationToken))
-            .WhereNotNull()
+            .Select((state, _) => ToMethodPartial(state.Left, state.Right))
             .Collect();
 
-    private static MethodPartial? ToMethodPartial(MethodDeclarationSyntax methodDeclaration, Compilation compilation, IncrementalGeneratorInitializationContext context, CancellationToken cancellationToken)
+    private static bool IsSyntaxTargetForGeneration(SyntaxNode node, CancellationToken cancellationToken)
+        => node is ClassDeclarationSyntax { AttributeLists.Count: > 0 };
+
+    private static SemanticTarget? GetSemanticTargetForGeneration(GeneratorSyntaxContext context, CancellationToken cancellationToken)
+        => context.Node is ClassDeclarationSyntax classDeclarationSyntax
+           && context.SemanticModel.GetDeclaredSymbol(classDeclarationSyntax, cancellationToken) is { } classSymbol
+           && classSymbol.GetAttributes()
+               .Where(a => a.AttributeClass?.ToDisplayString() == AttributeFullName)
+               .Where(AttributeBelongsToPartialPart(classDeclarationSyntax))
+               .Select(ParseAttribute)
+               .ToImmutableArray() is { Length: >=1 } attributes
+            ? new SemanticTarget(classDeclarationSyntax, attributes)
+            : null;
+
+    private static Func<AttributeData, bool> AttributeBelongsToPartialPart(ClassDeclarationSyntax partialPart)
+        => attribute => attribute.ApplicationSyntaxReference?.GetSyntax().Ancestors().OfType<ClassDeclarationSyntax>().FirstOrDefault() == partialPart;
+
+    private static ParsedAttribute ParseAttribute(AttributeData attribute)
+        => attribute.ConstructorArguments.Length >= 2
+           && attribute.ConstructorArguments[0].Value is INamedTypeSymbol type
```

Comment:
Maybe extract the magic numbers to a constant and use a descriptive name like "MethodNameIndex"

Reply:
[208eb18](https://github.com/polyadic/funcky/pull/634/commits/208eb188d0c3a6b0d069e2c8ee3db3a5993f14ca)